### PR TITLE
Fix for pecan getting stuck issue

### PR DESCRIPTION
**Pull request by mak-454**
@njagadish

### CODE REVIEW

**Comment by njagadish** [comment]
@hemanthravi

File: gbpservice/nfp/proxy_agent/proxy/proxy.py [hunk 1]
```diff
@@ -10,26 +10,29 @@
...
-import Queue
-from Queue import Empty
 import socket
 import sys
-import threading
 import time
 
 from oslo_log import log as logging
+from oslo_config import cfg as oslo_config
+from oslo_log import log as oslo_logging
```
Comment (njagadish) [inline gbpservice/nfp/proxy_agent/proxy/proxy.py:26]: Already imported oslo_log. Can't this be done the imported one.

File: gbpservice/nfp/proxy_agent/proxy/proxy.py [hunk 2]
```diff
@@ -135,29 +138,43 @@ def connect(self):
...
         self._tick()
         self._timedout()
 
     def idle_reset(self):
         self._idle_count = 0
+        self._start_time = time.time()
+
+    def _wait(self, timeout):
+        if self.type == 'unix':
+            eventlet.sleep(timeout)
```
Comment (njagadish) [inline gbpservice/nfp/proxy_agent/proxy/proxy.py:173]: The idea is to yield here. We are not looking for data for "timeout" seconds?

File: gbpservice/nfp/proxy_agent/proxy/proxy.py [hunk 3]
```diff
@@ -135,29 +138,43 @@ def connect(self):
...
     def idle(self):
         self._tick()
         self._timedout()
 
     def idle_reset(self):
         self._idle_count = 0
+        self._start_time = time.time()
+
+    def _wait(self, timeout):
+        if self.type == 'unix':
```
Comment (njagadish) [inline gbpservice/nfp/proxy_agent/proxy/proxy.py:172]: Why is the issue not for tcp? and only for unix socket?

**Comment by mak-454** [comment]
Raised a different PR on a different daily branch, got merged into base branch.

**Pull request closed without merging**